sql/core/src/main/scala/org/apache/spark/sql/execution/joins/SortMergeJoinExec.scala

```diff
@@ -251,7 +251,8 @@ case class SortMergeJoinExec(
               RowIterator.fromScala(rightIter),
               inMemoryThreshold,
               spillThreshold,
-              cleanupResources
+              cleanupResources,
+              condition.isEmpty
```

Comment:
Thanks @peter-toth !
I think this could be also added to LeftAnti join, which is also only interested in the existence of a match and doesn't need to buffer them.

Reply:
Thanks @juliuszsompolski, I think you are right. Shall I open a follow-up PR or a different ticket?

Reply:
I think a followup PR is fine.

Reply:
I've opened https://github.com/apache/spark/pull/29727